compositor_render/src/transformations/shader/pipeline.rs

```diff
@@ -10,24 +10,36 @@ use super::INPUT_TEXTURES_AMOUNT;
 
 pub struct Pipeline {
     pipeline: wgpu::RenderPipeline,
-    pub(super) bgls: BindGroupLayouts,
     geometry_buffers: RectangleRenderBuffers,
     sampler: Sampler,
+    pub(super) textures_bgl: wgpu::BindGroupLayout,
 }
 
 impl Pipeline {
-    pub fn new(device: &wgpu::Device, shader_source: wgpu::ShaderSource) -> Self {
-        let bgls = BindGroupLayouts::new(device);
-
+    pub fn new(
+        device: &wgpu::Device,
+        shader_source: wgpu::ShaderSource,
+        uniforms_bgl: &wgpu::BindGroupLayout,
+    ) -> Self {
         let sampler = Sampler::new(device);
 
+        let textures_bgl = device.create_bind_group_layout(&wgpu::BindGroupLayoutDescriptor {
+            label: Some("shader transformation textures bgl"),
+            entries: &[wgpu::BindGroupLayoutEntry {
+                binding: 0,
+                count: NonZeroU32::new(INPUT_TEXTURES_AMOUNT),
+                visibility: wgpu::ShaderStages::FRAGMENT | wgpu::ShaderStages::VERTEX,
+                ty: wgpu::BindingType::Texture {
+                    multisampled: false,
+                    sample_type: wgpu::TextureSampleType::Float { filterable: true },
+                    view_dimension: wgpu::TextureViewDimension::D2,
+                },
+            }],
+        });
+
         let pipeline_layout = device.create_pipeline_layout(&wgpu::PipelineLayoutDescriptor {
             label: Some("shader transformation pipeline layout"),
-            bind_group_layouts: &[
-                &bgls.textures_bgl,
-                &bgls.uniforms_bgl,
-                &sampler.bind_group_layout,
-            ],
+            bind_group_layouts: &[&textures_bgl, uniforms_bgl, &sampler.bind_group_layout],
```

Comment:
What if the shader doesn't require any parameters? We should allow to register shaders without any params. What do you think?

Reply:
I want to address that in a subsequent PR